src/middleware/jirahost-middleware.ts

```diff
@@ -19,10 +19,21 @@ const extractUnsafeJiraHost = (req: Request): string | undefined => {
 		return req.body?.jiraHost;
 	} else if (req.cookies.jiraHost) {
 		return req.cookies.jiraHost;
+	} else if (req.query?.jwt) {
+		return getJiraHostFromJwtToken(req.query.jwt as string);
 	}
 	return undefined;
 };
 
+const getJiraHostFromJwtToken = (encodedToken: string) => {
+	try {
+		const decodedToken = JSON.parse(Buffer.from(encodedToken.split(".")[1], "base64").toString());
+		return decodedToken.tenant_url;
+	} catch (error) {
+		return null;
```

Comment:
1. logging missing
2. Shouldn't this throw an error?

Reply:
I think, this will not set `jirahost` in cookies so subsequent calls such as repository search  or branch validation will fail